src/OrchardCore.Modules/OrchardCore.Users/Localization/Providers/UserLocalizationClaimsProvider.cs

```diff
@@ -0,0 +1,36 @@
+using System.Security.Claims;
+using OrchardCore.Users.Localization.Models;
+using OrchardCore.Entities;
+using OrchardCore.Users;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.Services;
+using System.Threading.Tasks;
+using System;
+
+namespace OrchardCore.Users.Localization.Providers;
+
+public class UserLocalizationClaimsProvider : IUserClaimsProvider
+{
+    public Task GenerateAsync(IUser user, ClaimsIdentity claims)
+    {
+        ArgumentNullException.ThrowIfNull(user);
+        ArgumentNullException.ThrowIfNull(claims);
+
+        if (user is not User u)
+        {
+            return Task.CompletedTask;
+        }
+
+        if (u.Has<UserLocalizationSettings>())
+        {
+            var localizationSetting = u.As<UserLocalizationSettings>();
+
+            if (!String.IsNullOrEmpty(localizationSetting.Culture))
```

Comment:
I don't think there's an issue if the user doesn't select a culture, then it will use invarient culture

Reply:
No, it will use the next RequestCultureProvider simply. No invariant culture unless it is a supported culture. The default value will be the system culture as a fallback like everything in ASP.NET.

Reply:
Yep, if it's supported

Reply:
Here I agree with @Skrypt, it just provides a culture claim if it exists, otherwise let things working as before.